src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SeparateSidewalkTagCheck.java

```diff
@@ -0,0 +1,343 @@
+     */
+    private boolean isCrossing(final PolyLine lineCrossed, final PolyLine crossingItem)
+    {
+        return !lineCrossed.intersections(crossingItem).isEmpty()
+                || lineCrossed.overlapsShapeOf(crossingItem)
+                || !this.isAcceptableHeading(
+                        Objects.requireNonNull(lineCrossed.overallHeading().orElse(null)),
+                        Objects.requireNonNull(crossingItem.overallHeading().orElse(null)));
+    }
+
+    /**
+     * Check if {@link Edge} is a dual carriageway road.
+     *
+     * @param edge
+     *            {@link Edge} to be checked
+     * @return {@code true} if {@link Edge} is dual carriageway.
+     */
+    private boolean isDualCarriageWay(final Edge edge)
+    {
+        final Optional<HighwayTag> highwayTag = HighwayTag.highwayTag(edge);
+        final Optional<OneWayTag> onewayTag = OneWayTag.tag(edge);
+        return highwayTag.isPresent() && onewayTag.isPresent()
+                && highwayTag.get().isMoreImportantThanOrEqualTo(this.maximumHighwayType)
+                && "YES".equals(onewayTag.get().toString());
+    }
+
+    /**
+     * Check if a location is to the right of a line.
+     *
+     * @param line
+     *            The line to compare the location to
+     * @param location
+     *            The location
+     * @return {@code true} if the location is to the right of the line.
+     */
+    private boolean isRightOf(final PolyLine line, final Location location)
+    {
+        final PolyLine tLine = new PolyLine(location);
+        final Segment closest = line.segments().stream()
+                .min(Comparator.comparingDouble(s -> s.shortestDistanceTo(tLine).asMeters()))
+                .orElse(null);
+        if (closest != null)
+        {
+            final PolyLine testLine = new PolyLine(closest.first(), closest.last(), location);
+            final Angle difference = testLine.headingDifference().orElse(null);
+            return difference != null && difference.asDegrees() > 0;
+        }
+        return false;
+    }
+
+    private boolean isSidewalkTaggingMismatch(final String sidewalkTagValue,
+            final int rightSidewalkCount, final int leftSidewalkCount)
+    {
+        switch (sidewalkTagValue)
+        {
+            // sidewalk=right but separate sidewalks on left or both sides are detected
+            case "right":
+                return rightSidewalkCount < 1 && leftSidewalkCount > 0
+                        || rightSidewalkCount > 0 && leftSidewalkCount > 0;
+            // sidewalk=left but separate sidewalks on right or both sides are detected
+            case "left":
+                return leftSidewalkCount < 1 && rightSidewalkCount > 0
+                        || leftSidewalkCount > 0 && rightSidewalkCount > 0;
+            // sidewalk=both but separate sidewalk on left or ride side is detected.
+            case "both":
+                return rightSidewalkCount < 1 || leftSidewalkCount < 1;
```

Comment:
I think this needs to check for and allow the case where right and left count are 0.
```suggestion
                return (rightSidewalkCount < 1 && leftSidewalkCount > 0)
                        || (rightSidewalkCount > 0 && leftSidewalkCount < 1);
```